src/character.cpp

```diff
@@ -4069,16 +4100,25 @@ void Character::set_stored_kcal( int kcal )
 void Character::set_stored_calories( int cal )
 {
     if( stored_calories != cal ) {
+        int cached_bmi = std::floor( get_bmi_fat() );
```

Comment:
you don't need to cache this if you're only using it once

Reply:
I believe it's necessary because the check calc_encumbrance(); is heavy enough that I don't want it to run every time stored calories updates. It records the whole-number BMI before the change, then compares the whole number BMI after the change to see if it's different because the limb encumbrance from obesity only cares about the whole number differences. I don't think there's a way to get BMI at an "artificial" kcal value, it only checks for your current kcal and you can't manipulate that in the function call. If that make sense